Sources/SwiftInspectorVisitors/FunctionDeclarationVisitor.swift

```diff
@@ -56,17 +56,21 @@ fileprivate final class FunctionSignatureVisitor: SyntaxVisitor {
       let type = node.type?.typeDescription
     else { return .skipChildren }
 
-    appendArgument(argumentLabelName: argumentLabelName, type: type)
+    let parameterName = node.secondName?.text
+
+    appendArgument(argumentLabelName: argumentLabelName, parameterName: parameterName, type: type)
     return .skipChildren
   }
   override func visit(_ node: ReturnClauseSyntax) -> SyntaxVisitorContinueKind {
     returnType = node.returnType.typeDescription
     return .skipChildren
   }
 
-  fileprivate func appendArgument(argumentLabelName: String, type: TypeDescription) {
+  fileprivate func appendArgument(argumentLabelName: String, parameterName: String?, type: TypeDescription) {
     var arguments = self.arguments ?? []
-    arguments.append(.init(argumentLabelName: argumentLabelName, type: type))
+    // By default, parameters use their parameter name as their argument label
+    let param = parameterName ?? argumentLabelName
+    arguments.append(.init(argumentLabelName: argumentLabelName, parameterName: param, type: type))
```

Comment:
I don't love the name `param`. I might write this as:
```suggestion
    // By default, parameters use their parameter name as their argument label
    let param = parameterName ?? argumentLabelName
    arguments.append(.init(
      argumentLabelName: argumentLabelName,
      parameterName: parameterName ?? argumentLabelName,
      type: type))
```
As an aside, the comment `By default, parameters use their parameter name as their argument label` doesn't make a whole lot of sense. The parameter name only falls back to the argument label if there is no explicit parameter name, right?

Reply:
I do think the comment `By default, parameters use their parameter name as their argument label` is correct per [this](https://docs.swift.org/swift-book/LanguageGuide/Functions.html#ID166)
<img width="808" alt="Screen Shot 2022-09-16 at 2 33 03 PM" src="https://user-images.githubusercontent.com/1791049/190787318-885023aa-8667-45f9-9f5d-37045efe2f7d.png">

Reply:
That's helpful context! Within that context, the comment makes sense. Putting that comment on top of the assignment of `param` is what confused me: we're setting the parameter name in the next line, not the argument label. And we're setting the parameter name to be the same as the argument label, rather than setting the argument label to be the same as the parameter name which is what the comment suggests we should be doing.

Maybe what's confusing me more is the fact that we're setting the `parameterName` to be `argumentLabelName` if there is no explicit `parameterName`. In fact, doing this loses data.

This code would create identical output for the following scanned methods:
```
func doSomething(parameterName parameterName: Void)
func doSomething(parameterName: Void)
```

Ideally the output of our scans could be used to re-create the original code (or at least the method signatures on the scanned types). This implementation doesn't let us do that with confidence

Reply:
> I don't love the name param. I might write this as:

I thought about doing that but wanted to take it out into its own line so I could add the code comment above. Since this is in the implementation and I _think_ param pretty clearly refers to "parameter" I think this is fine? But lemme know if you disagree!

Reply:
> Maybe what's confusing me more is the fact that we're setting the parameterName to be argumentLabelName if there is no explicit parameterName. In fact, doing this loses data.

I don't see a problem with scanning those 2 equally since they are, in fact, equal.

I think we've decided to be opinionated here in terms of style (e.g. using `Array` vs `[]`) so I think it's ok to lose the fact that the parameter name and argument labels are or are not explicitly defined.

Reply:
they're equal from a "how would you call the code" POV, but not from a "how would you generate a protocol for this method" POV. Though I just tried it out and the following code does compile:
```
protocol TestProtocol {
    func doSomething(parameterName parameterName: String)
}

final class TestConformance: TestProtocol {
    func doSomething(parameterName: String) {
        // do something
    }
}
```

So I withdraw the objection, since this API is not limiting a consumer's ability to generate compiling code.